bitnami/thanos/templates/bucketweb/deployment.yaml

```diff
@@ -30,13 +30,15 @@ spec:
     metadata:
       labels: {{- include "common.labels.standard" ( dict "customLabels" $podLabels "context" $ ) | nindent 8 }}
         app.kubernetes.io/component: bucketweb
+      {{- if or $.Values.bucketweb.podAnnotations (include "thanos.createObjstoreSecret" $) }}
       annotations:
-        {{- if (include "thanos.objstoreConfig" $) }}
-        checksum/objstore-configuration: {{ include "thanos.objstoreConfig" . | sha256sum }}
+        {{- if (include "thanos.createObjstoreSecret" $) }}
+        checksum/objstore-configuration: {{ include "thanos.objstoreConfig" $ | sha256sum }}
         {{- end }}
-        {{- if .Values.bucketweb.podAnnotations }}
+        {{- if $.Values.bucketweb.podAnnotations }}
```

Comment:
Not sure why there is a need to use root context (`$`) instead of current context (`.`) here. If I'm not mistaken there is no `range` context here.

This happens to most of the added logic. Only the `stateful-sharded.yaml` template should need the use of `$`.

Reply:
Hi, you are right, but $ doesn't break anything and to have it aligned I think it doesn't harm anything.

Reply:
I agree having `$` can be seen as a preventive measure but I'm still reticent to use it unless needed mainly because:

- Using `$` by default can make people think that `current_context != root_context`, which can cause confusion.
- By our styling convention, in `bitnami/charts` using `$` is reserved only for instances where the current context is not the same as the root context.

Reply:
Okay, clear, will have to change this.